monai/transforms/inverse.py

```diff
@@ -185,7 +185,12 @@ def track_transform_meta(
             # not lazy evaluation, directly update the metatensor affine (don't push to the stack)
             orig_affine = data_t.peek_pending_affine()
             orig_affine = convert_to_dst_type(orig_affine, affine, dtype=torch.float64)[0]
-            affine = orig_affine @ to_affine_nd(len(orig_affine) - 1, affine, dtype=torch.float64)
+            try:
+                affine = orig_affine @ to_affine_nd(len(orig_affine) - 1, affine, dtype=torch.float64)
+            except RuntimeError as e:
+                raise RuntimeError(
+                    "mismatch affine matrix, ensured that the batch dimension is not included in the calculation."
+                ) from e
```

Comment:
```suggestion
            except RuntimeError as e:
                if orig_affine.ndim > 2:
                    if data_t.is_batch:
                        msg = "Transform applied to batched tensor, should be applied to instances only"
                    else:
                        msg = "Mismatch affine matrix, ensured that the batch dimension is not included in the calculation."
                    raise RuntimeError(msg) from e
                else:
                    raise
```

We should only say something about this if we know the affine matrix has an odd dimensionality (and maybe should check of it's square or not). We can check a MetaTensor to see if it's a batched tensor this way and raise the exception like this in this case. For vanilla tensors the error message may be a bit confusing but this is still better. I haven't tried this code works so please double check if you want to use it.